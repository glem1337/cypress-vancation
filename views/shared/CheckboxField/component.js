import PropTypes from 'prop-types';

import Checkbox from '../Checkbox';

const CheckboxField = ({
  field,
  asComponent: Component,
  handleChange,
  ...props
}) => (
  <Component
    onChange={handleChange}
    checked={field.value}
    {...field}
    {...props}
  />
);

CheckboxField.defaultProps = {
  asComponent: Checkbox,
};

CheckboxField.propTypes = {
  asComponent: PropTypes.elementType,
  field: PropTypes.shape().isRequired,
  handleChange: PropTypes.func.isRequired,
};

export default CheckboxField;
