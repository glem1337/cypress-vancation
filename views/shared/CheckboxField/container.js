import React from 'react';

import PropTypes from 'prop-types';
import CheckboxFieldComponent from './component';

class CheckboxField extends React.PureComponent {
  static propTypes = {
    field: PropTypes.shape().isRequired,
    form: PropTypes.shape({
      setFieldValue: PropTypes.func.isRequired,
    }).isRequired,
  };

  handleChange = (e) => {
    const {
      field: { name },
      form: { setFieldValue },
    } = this.props;

    setFieldValue(name, e.target.value);
  };

  render = () => (
    <CheckboxFieldComponent {...this.props} handleChange={this.handleChange} />
  );
}

export default CheckboxField;
