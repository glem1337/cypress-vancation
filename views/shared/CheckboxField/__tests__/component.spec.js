import { shallow } from 'enzyme';

import CheckboxField from '../component';

describe('CheckboxField component matches snapshot', () => {
  const defaultProps = {
    field: { value: true, name: 'fieldName' },
    passThrough: 'prop',
    handleChange: jest.fn(),
  };

  it('with default props', () => {
    const wrapper = shallow(<CheckboxField {...defaultProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
