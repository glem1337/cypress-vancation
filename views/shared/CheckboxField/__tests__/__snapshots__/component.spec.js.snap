// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxField component matches snapshot with default props 1`] = `
<Checkbox
  checked={true}
  label={null}
  mixed={false}
  name="fieldName"
  onChange={[MockFunction]}
  passThrough="prop"
  value={true}
/>
`;
